pkg/chain/ethereum/ethers.go

```diff
@@ -0,0 +1,69 @@
+package ethereum
+
+import (
+	"fmt"
+	"math/big"
+	"regexp"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// Ethers is a custom type to handle Ether value parsing in configuration files
+// using BurntSushi/toml package. It supports wei, Gwei and ether units. The
+// Ether value is kept as `wei` and `wei` is the default unit.
+// The value can be provided in the text file as e.g.: `1 wei`, `200 Gwei` or
+// `0.5 ether`.
+type Ethers struct {
+	*big.Int
+}
+
+// The most common units for Ether values.
+const (
+	Wei Unit = iota
+	Gwei
+	Ether
+)
+
+// Unit represents Ether value unit.
+type Unit int
+
+func (u Unit) String() string {
+	return [...]string{"wei", "Gwei", "ether"}[u]
+}
+
+// UnmarshalText is a function used to parse a value of Ethers.
+func (e *Ethers) UnmarshalText(text []byte) error {
+	re := regexp.MustCompile(`^(\d*[\.]?[\d]*)[ ]*([\w]*)$`)
+	matched := re.FindSubmatch(text)
+
+	if len(matched) < 1 {
```

Comment:
Shouldn't it be `len(matched) < 3`? 

`FindSubmatch` at index `0` returns the leftmost match and anything to the right are the matches of subexpressions. For example:
`FindSubmatch("1 wei") = ["1 wei", "1", "wei"]`
`FindSubmatch("wei") = ["wei", "", "wei"]`
`FindSubmatch("") = ["", "", ""]`

Also, we are reading `matched[1]` and `matched[2]` later in this function so we assume they are there although it's not expressed in this check.

Reply:
We did `len(matched) < 1` to catch problems when text failed to be resolved with the regexp. But actually as we know what result our expression outputs we can even do `len(matched) != 3`.